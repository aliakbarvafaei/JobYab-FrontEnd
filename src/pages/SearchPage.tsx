import HeaderNewShort from "../components/Header/HeaderNewShort";
import TitlePages from "../components/TitlePages/TitlePages";
import Footer from "../components/Footer/Footer";
import SearchBox from "../components/SearchBox";
import { Button } from "@mui/material";
import SearchIcon from "@mui/icons-material/Search";
import SingleDropdownWithSearch from "../components/SingleDropdownWithSearch";
import Post from "../components/Post";
import CustomPagination from "../components/Pagination";
import { useHistory } from "react-router-dom";
import MobileMenu from "../components/MobileMenu/MobileMenu";
import { statesRedux } from "../ts/interfaces";
import { useSelector } from "react-redux";
import Header from "../components/ProfilesCompanies/Header";
import { useEffect, useState } from "react";
import { getJobTypes, getPrivatePosts, getStates } from "../services/api";
import { GeneralType, StateType } from "../constants/types";
import { parseURLParams } from "../services/utils/ParseURL";

const SearchPage: React.FC = () => {
  const history = useHistory();
  const { role } = useSelector((state: statesRedux) => state.userAuth);
  const [allPosts, setAllPosts] = useState([]);
  const [states, setStates] = useState<StateType[]>([]);
  const [jobTypes, setJobTypes] = useState<GeneralType[]>([]);

  const [category, setCategory] = useState<string>("");
  const [province, setProvince] = useState<string>("");
  const [counterPage, setcounterPage] = useState(1);
  const [searchInput, setSearchInput] = useState(() => {
    const parsedValue = parseURLParams(history.location.search);
    //@ts-ignore
    if (parsedValue && parsedValue.searchText?.[0]) {
      //@ts-ignore
      return parsedValue.searchText[0];
    } else return "";
  });
  useEffect(() => {
    // const filters: filtersInterface = {
    //   searchInput: searchInput,
    //   category: category ?? "",
    //   province: province ?? "",
    // };
    window.scrollTo({ top: 0, left: 0, behavior: "smooth" });
    getPrivatePosts(
      counterPage,
      6,
      searchInput,
      category === "همه دسته‌بندی‌ها" ? "" : category,
      province === "تمام استان‌ها" ? "" : province
    )
      .then((response) => {
        setAllPosts(response.data);
      })
      .catch((err) => {
        console.error(err);
      });
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
  useEffect(() => {
    getStates().then((data) => {
      setStates([
        { id: -1, title: "تمام استان‌ها", order: 5 },
        ...data.data.data,
      ]);
    });
  }, []);
  useEffect(() => {
    getJobTypes().then((data) => {
      setJobTypes([{ id: -1, title: "همه دسته‌بندی‌ها" }, ...data.data.data]);
    });
  }, []);
  return (
    <div>
      <MobileMenu />
      {role && role === "company" ? <Header /> : <HeaderNewShort />}
      <TitlePages title="جستجو" />
      <div className="md:mr-10 md:ml-10 xl:ml-3 xl:mr-3 xlmin:mr-20 xlmin:ml-20">
        <div
          className="flex lg:flex-col lgmin:flex-row"
          style={{
            marginTop: 50,
            background: "rgba(25, 118, 210, 0.3)",
            padding: 20,
            borderRadius: 16,
            gap: 20,
          }}
        >
          <SearchBox
            icon={<SearchIcon />}
            placeholder="عنوان شغلی، مهارت و..."
            value={searchInput}
            onChange={(event) => {
              setSearchInput(event.target.value);
            }}
          />
          <SingleDropdownWithSearch
            placeholder="استان را انتخاب کنید"
            options={states.map((item) => ({
              label: item.title,
              value: item.id,
            }))}
            //@ts-ignore
            value={states.find((item) => item.id === province)}
            onChange={(event) => {
              setProvince(event?.label.toString() ?? "");
            }}
            scrollColor="rgba(25, 118, 210, 0.3)"
          />
          <SingleDropdownWithSearch
            placeholder="دسته بندی را انتخاب کنید"
            options={jobTypes.map((item) => ({
              label: item.title,
              value: item.id,
            }))}
            //@ts-ignore
            value={jobTypes.find((item) => item.id === category)}
            onChange={(event) => {
              setCategory(event?.label.toString() ?? "");
            }}
            scrollColor="rgba(25, 118, 210, 0.3)"
          />
          <Button
            className="bg-[#ffe11b]"
            style={{
              background: "#ffe11b",
              paddingBlock: 10,
              paddingInline: 30,
              flexGrow: 1,
              fontSize: 18,
              fontWeight: "bold",
              color: "black ",
            }}
            onClick={() => {
              history.replace(
                searchInput ? `/search?searchText=${searchInput}` : `/search/`
              );
              
              getPrivatePosts(
                counterPage,
                6,
                searchInput,
                category === "همه دسته‌بندی‌ها" ? "" : category,
                province === "تمام استان‌ها" ? "" : province
              )
                .then((response) => {
                  setAllPosts(response.data);
                  
                })
                .catch((err) => {
                  console.error(err);
                });
              setcounterPage(1);
            }}
          >
            جستجو
          </Button>
        </div>
      </div>
      <div
        className="md:mr-6 md:ml-6 xl:ml-4 xl:mr-4 xlmin:mr-24 xlmin:ml-24"
        style={{
          display: "flex",
          justifyContent: "center",
        }}
      >
        <div
          className="xlmin:w-10/12 xl:w-full"
          style={{
            padding: 10,
            marginTop: "20px",
            gap: 20,
            display: "flex",
            flexDirection: "column",
            alignItems: "center",
          }}
        >
          {allPosts?.map((postDetail:any) => (
            <Post
              data={postDetail}
            />
          ))}
          {!!allPosts.length && (
            <CustomPagination
              count={allPosts.length < 6 ? 1 : allPosts.length % 6}
              page={counterPage}
              onChange={(_, value) => {
                setcounterPage(value);
              }}
            />
          )}
        </div>
      </div>
      <div style={{ marginTop: 40 }}>
        <Footer />
      </div>
    </div>
  );
};

export default SearchPage;
