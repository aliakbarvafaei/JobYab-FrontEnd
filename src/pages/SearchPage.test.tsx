import { screen, render } from "@testing-library/react";
import SearchPage from "./SearchPage";
import configureStore from "redux-mock-store";
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";

const mockStore = configureStore([]);
describe("Search Page", () => {
  let store: any;
  window.scrollTo = jest.fn();
  //   let history: any;

  beforeEach(() => {
    store = mockStore({
      userAuth: {
        role: "admin",
        token: "some-token",
      },
    });
  });
  test("check one property of fields", () => {
    render(
      <Router>
        <Provider store={store}>
          <SearchPage />
        </Provider>
      </Router>
    );
    const primaryButton = screen.getAllByRole("button")[1];
    expect(primaryButton).toHaveAttribute("aria-label", "حساب کاربری");
  });
  it("renders correctly in Search Page", () => {
    const { asFragment } = render(
      <Router>
        <Provider store={store}>
          <SearchPage />
        </Provider>
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  test("displays loading spinner while data is being fetched", () => {
    render(
      <Router>
        <Provider store={store}>
          <SearchPage />
        </Provider>
      </Router>
    );

    const loadingSpinner = screen.getByRole("progressbar");

    expect(loadingSpinner).toBeInTheDocument();
  });
  afterAll(() => {
    jest.clearAllMocks();
  });
});
