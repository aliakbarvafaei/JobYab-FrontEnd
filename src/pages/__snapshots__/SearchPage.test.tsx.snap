// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Page renders correctly in Search Page 1`] = `
<DocumentFragment>
  <div>
    <div
      class="sm:fixed smmin:hidden sm:bottom-0 z-[25] flex flex-row justify-between items-center bg-darkModeLightBlack w-full h-16 text-darkGray text-[26px] mmmin:px-[40px] mm:px-[20px]"
    >
      <a
        href="/profile?section=message"
      >
        <span
          class="flex flex-col justify-center items-center"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ContactSupportIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11.5 2C6.81 2 3 5.81 3 10.5S6.81 19 11.5 19h.5v3c4.86-2.34 8-7 8-11.5C20 5.81 16.19 2 11.5 2zm1 14.5h-2v-2h2v2zm0-3.5h-2c0-3.25 3-3 3-5 0-1.1-.9-2-2-2s-2 .9-2 2h-2c0-2.21 1.79-4 4-4s4 1.79 4 4c0 2.5-3 2.75-3 5z"
            />
          </svg>
          <p
            class="text-[12px] mm:text-[10px]"
          >
            پشتیبانی
          </p>
        </span>
      </a>
      <a
        href="/profile?section=bookmark"
      >
        <span
          class="flex flex-col justify-center items-center"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="BookmarksOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15 7v12.97l-4.21-1.81-.79-.34-.79.34L5 19.97V7h10m4-6H8.99C7.89 1 7 1.9 7 3h10c1.1 0 2 .9 2 2v13l2 1V3c0-1.1-.9-2-2-2zm-4 4H5c-1.1 0-2 .9-2 2v16l7-3 7 3V7c0-1.1-.9-2-2-2z"
            />
          </svg>
          <p
            class="text-[12px] mm:text-[10px]"
          >
            نشان شده‌ها
          </p>
        </span>
      </a>
      <a
        href="/profile-company"
      >
        <span
          class="flex flex-col justify-center items-center"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ApartmentIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M17 11V3H7v4H3v14h8v-4h2v4h8V11h-4zM7 19H5v-2h2v2zm0-4H5v-2h2v2zm0-4H5V9h2v2zm4 4H9v-2h2v2zm0-4H9V9h2v2zm0-4H9V5h2v2zm4 8h-2v-2h2v2zm0-4h-2V9h2v2zm0-4h-2V5h2v2zm4 12h-2v-2h2v2zm0-4h-2v-2h2v2z"
            />
          </svg>
          <p
            class="text-[12px] mm:text-[10px]"
          >
            کارفرما
          </p>
        </span>
      </a>
      <span
        class="flex flex-col justify-center items-center"
      >
        <i
          aria-hidden="true"
          class="fa fa-user group relative"
        >
          <div
            class="bg-darkModeLightBlack absolute hidden rounded-md group-hover:block hover:flex w-[100px] py-[10px] px-[20px] bottom-6 right-[-45px]
                  flex-col drop-shadow-lg z-[26]"
          >
            <div
              class="text-right text-[14px] py-[12px] hoverItem font-normal cursor-pointer MuiBox-root css-0"
            >
              درخواست‌ها
            </div>
            <div
              class="text-right text-[14px] py-[12px] hoverItem font-normal cursor-pointer MuiBox-root css-0"
            >
              اطلاعات
            </div>
            <div
              class="text-right text-[14px] py-[12px] hoverItem font-normal cursor-pointer MuiBox-root css-0"
            >
              خروج
            </div>
          </div>
        </i>
        <p
          class="text-[12px] mm:text-[10px]"
        >
          حساب
        </p>
      </span>
    </div>
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic lg:px-[20px] lgmin:px-[80px] css-hip9hq-MuiPaper-root-MuiAppBar-root"
      style="min-height: 90px; justify-content: center; font-family: IRANSans;"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular css-r6ewbb-MuiToolbar-root"
          style="display: flex; justify-content: space-between;"
        >
          <a
            class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap md:hidden mdmin:flex css-8u39c-MuiTypography-root"
            href="/"
            style="font-family: monospace; font-weight: 700; letter-spacing: .3rem; text-decoration: none;"
          >
            JOBYAB
          </a>
          <div
            class="md:!flex mdmin:!hidden MuiBox-root css-0"
            style="flex-grow: 1;"
          >
            <button
              aria-controls="menu-appbar"
              aria-haspopup="true"
              aria-label="account of current user"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeLarge p-0 css-1pz7awu-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="MenuIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <a
            class="MuiTypography-root MuiTypography-h5 MuiTypography-noWrap md:flex mdmin:hidden md:text-[16px] mdmin:text-[20px] css-1961ak2-MuiTypography-root"
            href="/"
            style="flex-grow: 1; font-family: monospace; font-weight: 700; letter-spacing: .3rem; text-decoration: none;"
          >
            JOBYAB
          </a>
          <div
            class="MuiBox-root css-0"
            style="flex-grow: 0; align-items: center;"
          >
            <a
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium md:hidden mdmin:inline css-sghohy-MuiButtonBase-root-MuiButton-root"
              href="/profile-company"
              style="border-radius: 5px; color: black; background-color: white; gap: 5px; padding: 5px 20px 5px 20px; font-family: IRANSans; margin-left: 20px; margin-right: 20px;"
              tabindex="0"
            >
              بخش کارفرما
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </a>
            <a
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium md:hidden mdmin:inline-flex css-1e6y48t-MuiButtonBase-root-MuiButton-root"
              href="/profile?section=bookmark"
              style="margin-top: 2px; margin-bottom: 2px; color: white; gap: 5px; padding-left: 20px; font-family: IRANSans;"
              tabindex="0"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="BookmarksOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15 7v12.97l-4.21-1.81-.79-.34-.79.34L5 19.97V7h10m4-6H8.99C7.89 1 7 1.9 7 3h10c1.1 0 2 .9 2 2v13l2 1V3c0-1.1-.9-2-2-2zm-4 4H5c-1.1 0-2 .9-2 2v16l7-3 7 3V7c0-1.1-.9-2-2-2z"
                />
              </svg>
              نشان شده‌ها
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </a>
            <a
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium md:hidden mdmin:inline-flex css-1e6y48t-MuiButtonBase-root-MuiButton-root"
              href="/profile?section=message"
              style="margin-top: 2px; margin-bottom: 2px; color: white; gap: 5px; padding-left: 20px; font-family: IRANSans;"
              tabindex="0"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ContactSupportIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11.5 2C6.81 2 3 5.81 3 10.5S6.81 19 11.5 19h.5v3c4.86-2.34 8-7 8-11.5C20 5.81 16.19 2 11.5 2zm1 14.5h-2v-2h2v2zm0-3.5h-2c0-3.25 3-3 3-5 0-1.1-.9-2-2-2s-2 .9-2 2h-2c0-2.21 1.79-4 4-4s4 1.79 4 4c0 2.5-3 2.75-3 5z"
                />
              </svg>
              پشتیبانی
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </a>
            <button
              aria-label="حساب کاربری"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-yvetkp-MuiButtonBase-root-MuiIconButton-root"
              data-mui-internal-clone-element="true"
              style="padding: 0px;"
              tabindex="0"
              type="button"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circular md:w-[30px] md:h-[30px] md:text-[0.8rem] mdmin:w-[40px] mdmin:h-[40px] mdmin:text-[1.25rem] css-1wlk0hk-MuiAvatar-root"
              >
                <img
                  class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                  src="/static/images/avatar/2.jpg"
                />
              </div>
              <div
                class="text-[14px] pr-2 text-white md:hidden"
              >
                حساب کاربری
              </div>
            </button>
          </div>
        </div>
      </div>
    </header>
    <div
      class="bg-lightGray text-black mdmin:py-[25px] md:py-[10px] px-total bg-lightGray"
    >
      <div
        class="flex mdmin:flex-row md:flex-col justify-between"
      >
        <div
          class="text-[14px] md:pt-[10px] font-black text-center"
        >
          <span
            class="cursor-pointer"
          >
            خانه 
          </span>
          / جستجو 
        </div>
      </div>
    </div>
    <div
      class="flex justify-center mt-20"
    >
      <span
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </span>
    </div>
    <div
      style="margin-top: 40px;"
    >
      <section>
        <div
          class="bg-lightGray lg:pb-[100px] md:pt-[20px] lg:pt-[30px] lgmin:py-[45px] px-total bg-lightGray"
          data-testid="main-footer"
        >
          <div
            class="flex flex-row flex-wrap justify-between"
          >
            <div
              class="md:w-[100%] lg:w-[45%] lgmin:w-[30%]"
            >
              <a
                class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-8u39c-MuiTypography-root"
                href="/"
                style="display: flex; justify-content: center; font-family: monospace; font-weight: 700; letter-spacing: .3rem; text-decoration: none;"
              >
                JOBYAB
              </a>
              <p
                class="pt-[18px] text-center px-[40px] text-darkGray text-[14px] leading-[30px]"
              >
                ما در جاب یاب به شما کمک می‌کنیم تا بتوانید از فرصت های شغلی مختلف باخبر باشید و براحتی در جاب یاب صاحب کار شوید.
              </p>
              <div
                class="flex flex-row justify-between items-center mt-[8px]"
              >
                <span
                  class="cursor-pointer"
                  data-testid="facebook-icon"
                >
                  <svg
                    fill="currentColor"
                    font-size="20px"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 320 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
                    />
                  </svg>
                </span>
                <span
                  class="cursor-pointer"
                  data-testid="google-plus-icon"
                >
                  <svg
                    fill="currentColor"
                    font-size="20px"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 640 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M386.061 228.496c1.834 9.692 3.143 19.384 3.143 31.956C389.204 370.205 315.599 448 204.8 448c-106.084 0-192-85.915-192-192s85.916-192 192-192c51.864 0 95.083 18.859 128.611 50.292l-52.126 50.03c-14.145-13.621-39.028-29.599-76.485-29.599-65.484 0-118.92 54.221-118.92 121.277 0 67.056 53.436 121.277 118.92 121.277 75.961 0 104.513-54.745 108.965-82.773H204.8v-66.009h181.261zm185.406 6.437V179.2h-56.001v55.733h-55.733v56.001h55.733v55.733h56.001v-55.733H627.2v-56.001h-55.733z"
                    />
                  </svg>
                </span>
                <span
                  class="cursor-pointer"
                  data-testid="twitter-icon"
                >
                  <svg
                    fill="currentColor"
                    font-size="20px"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
                    />
                  </svg>
                </span>
                <span
                  class="cursor-pointer"
                  data-testid="instagram-icon"
                >
                  <svg
                    fill="currentColor"
                    font-size="20px"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 448 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
                    />
                  </svg>
                </span>
                <span
                  class="cursor-pointer"
                  data-testid="wifi-icon"
                >
                  <svg
                    fill="currentColor"
                    font-size="20px"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 640 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M634.91 154.88C457.74-8.99 182.19-8.93 5.09 154.88c-6.66 6.16-6.79 16.59-.35 22.98l34.24 33.97c6.14 6.1 16.02 6.23 22.4.38 145.92-133.68 371.3-133.71 517.25 0 6.38 5.85 16.26 5.71 22.4-.38l34.24-33.97c6.43-6.39 6.3-16.82-.36-22.98zM320 352c-35.35 0-64 28.65-64 64s28.65 64 64 64 64-28.65 64-64-28.65-64-64-64zm202.67-83.59c-115.26-101.93-290.21-101.82-405.34 0-6.9 6.1-7.12 16.69-.57 23.15l34.44 33.99c6 5.92 15.66 6.32 22.05.8 83.95-72.57 209.74-72.41 293.49 0 6.39 5.52 16.05 5.13 22.05-.8l34.44-33.99c6.56-6.46 6.33-17.06-.56-23.15z"
                    />
                  </svg>
                </span>
              </div>
            </div>
            <div
              class="mt-[30px] md:w-[100%] lg:w-[45%] lgmin:w-[20%] md:mt-[15px]"
            >
              <h4
                class="font-black text-[16px] mb-[25px] lg:mb-[15px] md:pb-[8px] md:mb-0 md:border-b-[1px] border-darkModeGray md:border-solid"
              >
                حساب کاربری
              </h4>
              <ul
                class="list-none p-0 text-[16px]"
              >
                <a
                  href="/profile-company?section=mypost"
                >
                  <li
                    class="relative inline-block pt-[13px] text-darkGray hoverItem"
                  >
                    آگهی های من
                  </li>
                </a>
                <div />
                <a
                  href="/profile"
                >
                  <li
                    class="relative inline-block pt-[13px] text-darkGray hoverItem"
                  >
                    پروفایل
                  </li>
                </a>
                <div />
              </ul>
            </div>
            <div
              class="mt-[30px] md:w-[100%] lg:w-[45%] lgmin:w-[20%] lg:mt-[15px]"
            >
              <h4
                class="font-black text-[16px] mb-[25px] lg:mb-[15px] md:pb-[8px] md:mb-0 md:border-b-[1px] border-darkModeGray md:border-solid"
              >
                آگهی
              </h4>
              <ul
                class="list-none p-0 text-[16px]"
              >
                <a
                  href="/profile?section=bookmark"
                >
                  <li
                    class="relative inline-block pt-[13px] text-darkGray hoverItem"
                  >
                    نشان شده‌ها
                  </li>
                </a>
                <div />
                <div>
                  <li
                    class="relative inline-block pt-[13px] text-darkGray hoverItem cursor-pointer"
                  >
                    فوری
                  </li>
                </div>
                <div />
              </ul>
            </div>
            <div
              class="mt-[30px] md:w-[100%] lg:w-[45%] lgmin:w-[20%] lg:mt-[15px]"
            >
              <h4
                class="font-black text-[16px] mb-[25px] lg:mb-[15px] md:pb-[8px] md:mb-0 md:border-b-[1px] border-darkModeGray md:border-solid"
              >
                درباره ما
              </h4>
              <ul
                class="list-none p-0 text-[16px]"
              >
                <li
                  class="pl-[25px] relative pt-[13px] text-darkGray block hoverItem"
                >
                  <i
                    aria-hidden="true"
                    class="fa fa-map-marker absolute left-0 top-[16px]"
                  />
                   اصفهان، دانشگاه اصفهان
                </li>
                <li
                  class="pl-[25px] relative pt-[13px] text-darkGray block hoverItem"
                >
                  <i
                    aria-hidden="true"
                    class="fa fa-phone absolute left-0 top-[16px]"
                  />
                   تلفن : 09123456789
                </li>
                <li
                  class="pl-[25px] relative pt-[13px] text-darkGray block hoverItem"
                >
                  <i
                    aria-hidden="true"
                    class="fa fa-envelope absolute left-0 top-[16px]"
                  />
                   ایمیل : jobyab@gmail.com
                </li>
              </ul>
            </div>
          </div>
        </div>
      </section>
    </div>
  </div>
</DocumentFragment>
`;
